sdk/core/core-rest-pipeline/review/core-rest-pipeline.api.md

```diff
@@ -265,6 +278,7 @@ export interface PipelineResponse {
 
 // @public
 export interface PipelineRetryOptions {
+    failoverHostGenerator?: FailoverHostGenerator;
```

Comment:
Do we want to expose this to consumers? I think it's unlikely a dev will want to customize the way failover works for a client. 

Since we don't really have a way to hide particular properties from our already public interfaces, I wonder if we should create an `internalRetryOptions` for this?

Reply:
The only situation I can think of is if a user were to roll their own replication stack for a service that offers <= 1 replica. Maybe to optimize latency with geolocation... without using a CDN... Do you think that use case is plausible enough for us to support it?

Reply:
I think if such an advanced user exists, they would be comfortable doing some casting tricks to pass the option without us officially exposing it on clients. Mostly I'm looking to avoid committing to a particular public shape here until we can get some real-world usage via SDKs